Diagnostics/FreeBusyChecker/FreeBusyChecker.ps1

```diff
@@ -0,0 +1,311 @@
+Show help for this script.
+
+.EXAMPLE
+.\FreeBusyChecker.ps1
+This cmdlet will run the Free Busy Checker script and Check Availability OAuth and DAuth Configurations both for Exchange On-Premises and Exchange Online.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Auth OAuth
+This cmdlet will run the Free Busy Checker Script against OAuth Availability Configurations.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Auth DAuth
+This cmdlet will run the Free Busy Checker Script against DAuth Availability Configurations.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Org ExchangeOnline
+This cmdlet will run the Free Busy Checker Script for Exchange Online Availability Configurations.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Org ExchangeOnPremise
+This cmdlet will run the Free Busy Checker Script for Exchange On-Premises OAuth or DAuth Availability Configurations.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Org All
+This cmdlet will run the Free Busy Checker Script for Exchange On-Premises and Exchange Online OAuth or DAuth Availability Configurations.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Org ExchangeOnPremise -Auth OAuth
+This cmdlet will run the Free Busy Checker Script for Exchange On-Premises Availability OAuth Configurations
+#>
+
+# Exchange On-Premises
+#>
+#region Properties and Parameters
+
+#Requires -Module ExchangeOnlineManagement
+#Requires -Module ActiveDirectory
+
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', '', Justification = 'Variables are being used')]
+[CmdletBinding(DefaultParameterSetName = "FreeBusyInfo_OP", SupportsShouldProcess)]
+
+param(
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [ValidateSet('DAuth', 'OAuth', 'All', '')]
+    [string[]]$Auth,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [ValidateSet('ExchangeOnPremise', 'ExchangeOnline')]
+    [string[]]$Org,
+    [Parameter(Mandatory = $true, ParameterSetName = "Help")]
+    [switch]$Help,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremisesUser,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnlineUser,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremDomain,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremEWSUrl,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremLocalDomain
+)
+begin {
+    . $PSScriptRoot\Functions\OnPremDAuthFunctions.ps1
+    . $PSScriptRoot\Functions\OnPremOAuthFunctions.ps1
+    . $PSScriptRoot\Functions\ExoDAuthFunctions.ps1
+    . $PSScriptRoot\Functions\ExoOAuthFunctions.ps1
+    . $PSScriptRoot\Functions\htmlContent.ps1
+    . $PSScriptRoot\Functions\hostOutput.ps1
+    . $PSScriptRoot\Functions\CommonFunctions.ps1
+} end {
+    $countOrgRelIssues = (0)
+    $Script:FedTrust = $null
+    $Script:AutoDiscoveryVirtualDirectory = $null
+    $Script:OrgRel = $null
+    $Script:SPDomainsOnprem = $null
+    $AvailabilityAddressSpace = $null
```

Comment:
This isn't required to be here as you set it in the other functions and only use it there.

Reply:
Removed